src/python/lib/ensembl/compara/utils/cigar.py

```diff
@@ -14,9 +14,57 @@
 # limitations under the License.
 """Collection of utils methods targeted to cigar line manipulation."""
 
-__all__ = ['aligned_seq_to_cigar', 'alignment_to_seq_coordinate', 'get_cigar_array']
-
-from typing import List, Tuple
+__all__ = ['aligned_seq_to_cigar', 'alignment_to_seq_coordinate',
+           'get_cigar_array', 'alignment_to_seq_region']
+
+from typing import List, Tuple, Optional
+
+def alignment_to_seq_region(cigar: str, start: int, end: int) -> Optional[Tuple[int,int]]:
+    """Convert the coordinate of a region from the alignment level to the unaligned seq level
+
+         An ungapped with a start and an end in a non-gaped area will return a direct coordinate mapping i.e
+         the region with a start = 3 and an end = 12 in the aligned sequence below will return (3,7).
+         A region with start and end in the single gap opening will return an empty pair as the region
+         does not exist in this sequence, i.e a region with start=5 and end=8 will return ()
+         A region with a start in a gap opening will return the first position after the gap closes whether
+         the end is in an alternative gapped or ungapped position ie start = 6 and end = 14 will return (5,9).
+         A region with an end in a gap will return the last position before the gap whether the start is
+         in an ungapped or alternative gap position. i.e start = 6 and end = 20 will return (5,9)
+
+        aligned:     1  2  3  4  5  6  7  8  9  10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26
+                     A  A  T  C  -  -  -  -  -  A  C  T  C  T  -  -  -  -  -  -  -  -  T  C  T  C
+        unaligned:   1  2  3  4                 5  6  7  8  9                          10 11 12 13
+
+
+        Args:
+            cigar: cigar line of the aligned sequence
+            start: start coordinate of the region in the alignment
+            end: end coordinate of the region in the alignment
+        Returns:
+            coordinate of the unaligned region at the sequence level, empty tuple if the region is
+            inside a gap
```

Comment:
```suggestion
```
This isn't necessary/is redundant as it is already explained in opening paragraph.